components/NaveSlider.vue

```diff
@@ -0,0 +1,222 @@
+    </div>
+  </div>
+</template>
+
+<script>
+import smoothscroll from 'smoothscroll-polyfill'
+
+export default {
+  props: {
+    expanded: {
+      type: Boolean
+    },
+
+    animationOnScroll: {
+      type: Boolean,
+      default: true
+    },
+
+    sliderSlug: {
+      type: String,
+      default: ''
+    }
+  },
+
+  data () {
+    return {
+      start: 1,
+      difference: 0,
+      drag: false,
+      hideBar: false,
+      element: null,
+      hiddenArrow: false,
+      position: 0,
+      lastCardChild: null
+    }
+  },
+
+  computed: {
+    expandedClass () {
+      return this.expanded ? 'nave-slider--expanded' : 'full-width'
+    },
+
+    expandedContainerClass () {
+      return this.expanded ? 'nave-slider__container--expanded' : ''
+    }
+  },
+
+  mounted () {
+    this.getPosition()
+    this.lastCardChild = this.$refs.sliderContent.lastChild
+    this.element = this.$refs.slider
+
+    window.__forceSmoothScrollPolyfill__ = true
+    smoothscroll.polyfill()
+    window.addEventListener('mouseup', this.mouseUp)
+  },
+
+  destroyed () {
+    window.removeEventListener('mouseup', this.mouseUp)
+  },
+
+  methods: {
+    mouseDown (event) {
+      if (!event) {
+        event = window.event
+      }
+
+      event.preventDefault()
+      this.start = event.clientX + this.element.scrollLeft
+      this.difference = 0
+      this.drag = true
+    },
+
+    mouseMove (event) {
+      if (!this.drag) {
+        return null
+      }
+
+      // this.hiddenScrollArrow()
+
+      if (!event) {
+        event = window.event
+      }
+
+      event.preventDefault()
+      this.difference = this.start - (event.clientX + this.element.scrollLeft)
+      this.element.scrollLeft += this.difference
+      this.hiddenScrollArrow()
+    },
+
+    mouseUp (event) {
+      let tick = 1
+
+      if (!event) {
+        event = window.event
+      }
+
+      event.preventDefault()
+      this.drag = false
+
+      const animate = () => {
+        const step = Math.sin(tick)
+
+        if (step <= 0) {
+          this.difference = 0
+          window.cancelAnimationFrame(animate)
+        } else {
+          this.element.scrollLeft += this.difference * step
+          tick -= 0.02
+          window.requestAnimationFrame(animate)
+        }
+      }
+
+      this.animationOnScroll && animate()
+    },
+
+    getPosition () {
+      let offsetTrail = this.$refs.sliderContent.lastChild
+      let offsetLeft = 0
+      while (offsetTrail) {
+        offsetLeft += offsetTrail.offsetLeft
+        offsetTrail = offsetTrail.offsetParent
+      }
+
+      this.position = offsetLeft
+    },
+
+    next () {
+      this.element.scroll({
+        left: this.element.scrollLeft + this.lastCardChild.offsetWidth,
+        behavior: 'smooth'
+      })
+
+      this.hiddenScrollArrow()
+    },
+
+    hiddenScrollArrow () {
+      const hidden = window.innerWidth + this.element.scrollLeft
+
+      if (hidden < this.position) {
```

Comment:
`this.hiddenArrow = hidden > this.position` isso já n resolve tudo?

Reply:
Mágico, funcionou sim